src/frontend/src/lib.rs

```diff
@@ -168,8 +171,22 @@ pub fn start(opts: FrontendOpts) -> Pin<Box<dyn Future<Output = ()> + Send>> {
     Box::pin(async move {
         let listen_addr = opts.listen_addr.clone();
         let session_mgr = Arc::new(SessionManagerImpl::new(opts).await.unwrap());
-        pg_serve(&listen_addr, session_mgr, TlsConfig::new_default())
-            .await
-            .unwrap();
+        let redact_sql_option_keywords = Arc::new(
+            session_mgr
+                .env()
+                .batch_config()
+                .redact_sql_option_keywords
+                .iter()
+                .map(|s| s.to_lowercase())
+                .collect::<HashSet<_>>(),
+        );
+        pg_serve(
+            &listen_addr,
+            session_mgr,
+            TlsConfig::new_default(),
+            Some(redact_sql_option_keywords),
+        )
```

Comment:
This looks a little bit dirty though. 🤣

Reply:
Yes...
I also thought about using another task_local for `redact_sql_option_keywords`, to avoid passing it through down the call chain. But it would reduce the maintainability.

Any other suggestions?